src/Sulu/Bundle/CategoryBundle/Domain/Event/CategoryCreatedEvent.php

```diff
@@ -72,18 +72,23 @@ public function getResourceId(): string
         return (string) $this->category->getId();
     }
 
-    public function getResourceLocale(): ?string
+    public function getResourceLocale(): string
     {
         return $this->locale;
     }
 
     public function getResourceTitle(): ?string
     {
-        $translation = $this->category->findTranslationByLocale($this->locale);
+        $translation = $this->category->findTranslationByLocale($this->getResourceTitleLocale());
 
         return $translation ? $translation->getTranslation() : null;
     }
 
+    public function getResourceTitleLocale(): string
+    {
+        return $this->getResourceLocale();
+    }
```

Comment:
would move this as default implementation into the `DomainEvent` and only overwrite the method in the event classes that need another implementation